### PR TITLE
retry conda install

### PR DESCRIPTION
**Pull request by tomkinsc**
add retry option to run_and_print; this is to support retrying conda
installs when anaconda.org errors out

### CODE REVIEW

File: travis/install-tools.sh [hunk 1]
```diff
@@ -36,7 +36,14 @@ export JAVA_HOME="$(pwd)/tools/conda-tools/default/jre"
 
 echo "Installing and validating bioinformatic tools"
 export CONDA_ENVS_PATH=tools/conda-cache:tools/conda-tools/default
-conda create -y -m -c bioconda -p tools/conda-tools/default --file $HOME/requirements-conda.txt python="$TRAVIS_PYTHON_VERSION"
+
+tries=0
+until [ $tries -ge 4 ]
```
Comment (yesimon) [inline travis/install-tools.sh]: for i in $(seq 3)

File: util/misc.py [hunk 2]
```diff
@@ -198,80 +199,91 @@ def run(args, stdin=None, stdout=None, stderr=None, shell=False,
 def run_and_print(args, stdout=None, stderr=None,
                   stdin=None, shell=False, env=None, cwd=None,
                   timeout=None, silent=False, buffered=False, check=False,
-                  loglevel=None):
+                  loglevel=None, expected_return_code=0, retries=0, retry_delay_sec=2):
```
Comment (yesimon) [inline util/misc.py]: So we should have an `assert not (check and retries)` ?

Comment (yesimon) [inline util/misc.py]: Also need docs that expected_return_code is only for retries?

**Comment by dpark01** [comment]
(unless I'm misunderstanding how the problem is manifesting itself...)

**Pull request closed without merging**